### ISSUE TITLE
if key is not a unicode string then set/get fails

### ISSUE THREAD

**Issue by farshidce** [jaysonsantos/python-binary-memcached#95]
I am using python-binary-memcached ( version 0.18) which did not distinguish between a string or a non-string and hence I am using it to insert binary keys. the way the key is set in my case is that the key which is a string is mspacked and then is compressed using lzf.

there are 5 places where encode function fails and I have listed two as an example.
I can submit a patch to fix this but wanted to check here and see what is the proposed solution.

the way I can think of solving this is that if value.encode() or value.encode("utf8") to decode fails then we can pass the value as it is instead of failing the user.

but if this library now insists on supporting string keys then perhaps we can add a new config that the user can set in the init method which by default will disable these encode/decode functions

1- /client.py", line 143,
  File "/usr/local/lib/python2.7/site-packages/bmemcached/client.py", line 143, in get_multi
    results = server.get_multi(keys)
  File "/usr/local/lib/python2.7/site-packages/bmemcached/protocol.py", line 471, in get_multi
    d[key.decode()] = self.deserialize(value, flags), cas
UnicodeDecodeError: 'ascii' codec can't decode byte 0xb1 in position 1: ordinal not in range(128)
2-protocol.py", line 457
  File "/usr/local/lib/python2.7/site-packages/bmemcached/protocol.py", line 457, in get_multi
    len(last), 0, 0, 0, len(last), 0, 0, last.encode())

**Comment by jaysonsantos** [comment]
Hey @farshidce, thanks for your report. I would like to keep it the most compatible as possible with pylibmc, do you know what is the behavior there? If you want to send a patch that makes it compatible to pylibmc it will be appreciated.

**Comment by farshidce** [comment]
yes. @jaysonsantos . I am able to read/write these binary keys using pylibmc ( and also using older version or python-binary-memcached. )

latest version of bmemcached:
```
import struct
import bmemached
packed = struct.pack("<Q", x)  
bmc = bmemcached.Client()
value = bmc.set(packed, "binary key inserted with bmemcached")
print(value)

      7     """Simply convert a string type to bytes."""
      8     if isinstance(value, six.string_types):
----> 9         return value.encode()
     10 
     11     return value

UnicodeDecodeError: 'ascii' codec can't decode byte 0x92 in position 0: ordinal not in range(128)

```

and using latest version pylibmc
```
import struct
import pylibmc
mc = pylibmc.Client(["127.0.0.1"], binary=True)
mc.set(packed, "binary key inserted with pylibmc")
value = mc.get(packed)
print(value)

In [42]: 
binary key inserted with pylibmc

```

I will send you a pull request to support binary keys in the same fashion that pylibmc does then.

**Comment by jaysonsantos** [comment]
Yeah, that was because I put compatibility with python 3. You can send the
PR to make it compatible with pylibmc.
Thanks!

**Comment by jaysonsantos** [comment]
fixed by #96
